.github/workflows/build.yml

```
@@ -16,8 +16,15 @@ name: build
 on: ['push']
```

This is not related to your change. I notice that the CI is trying to push to the container registry in the PR CI run. This is a huge problem. It should only push to the container registry when there is a push to the main branch, and main branch only.
```suggestion
on:
  push:
    branches:
      - 'develop'
```
```suggestion
on: ['push']
```